### DISCUSSION TITLE
Import business process catalog in DevOps

### DISCUSSION THREAD

**Discussion by ClaraSeijoBlanque** [microsoft/dynamics365patternspractices#110] [Q&A] [answered]
Hi Rachel and team,

I would need some help with the DevOps upload because I'm struggling a lot with the format. These are the steps I'm following:

- I downloaded the file "Full business process catalog ADO upload Oct 23".
- Once I open the file in Excel and I convert the text in columns separate by comma, there are 21 columns.

![image](https://github.com/microsoft/dynamics365patternspractices/assets/131162800/31acfdb1-a4be-4d85-927f-cdb0b3106e1e)

- But I cannot upload the file in DevOps because of the columns. If I create a work item manually in DevOps and I export the file from the query, the file only has 14 columns and short in other way.

![image](https://github.com/microsoft/dynamics365patternspractices/assets/131162800/4fe9bddb-847c-451b-a271-e56c78010a3e)

For example, the column "Title" exists 3 times in the original document but in DevOps this is a unique field.

So, the only way I can upload the file is if I copy/paste the columns in the DevOps template and, example for Title I combine the 3 existing columns in 1. That means a lot of manual work and data transformation.

Could you please help me to understand if I'm using the correct file for DevOps?

Thanks in advance,
Clara

**Comment by ghost** [comment]
Hello Clara,

The different title columns are necessary to insert the parent/child structure. yon can add them at will in your CSV if you want to a level.

As for columns, for them to appear in the extract, they must be part of the view or query you are generating the CSV from. Use the "column options" to show them.

Having that said, i think some of the fields in the template are custom fields, like Process sequence ID, Business outcome strategy. Additionally, some configuration is required before loading the file, like the area path. Area path probably needs to be renamed to match your target project.

Hope it helps.

**Reply by ClaraSeijoBlanque** [reply]
Hi Alexander, thanks for your answer. 

I understand the reason to have multiple Title columns to have the structure you mentioned, but this is not possible in DevOps, that's the issue, there the title is unique and the work items are created with the different types (epic, feature, user story..). That's why for example only for title I need to be modifying the template to be able to uploaded.

Following the article that Eva mentioned I understand how to manage "Area path" for example, not issue with that, this is an easy "replace" in the excel. And I can see that some columns are custom and are not included in DevOps, that's OK.

The main issue is to be moving data from one column to other column without lost info.

Hope this clarify the doubt. Thanks again
Clara

**Comment by edupont04** [comment]
We'll publish an article soon with suggestions for how to use the Excel file, but you can already now find some guidance in the import-business-processes-ADO.md file at [https://github.com/microsoft/dynamics365patternspractices/blob/main/templates/business-processes/](https://github.com/microsoft/dynamics365patternspractices/blob/main/templates/business-processes/import-business-processes-ADO.md) .

**Reply by ClaraSeijoBlanque** [reply]
Thanks Eva. I follow up these steps, this is really usual, but it doesn't explains the issues I'm finding. I will wait for the article you mention.

**Reply by rachel-profitt** [reply, marked as answer]
Hi Clara, 
Can you provide a screenshot of the error you are receiving when you attempt to import the file provided? The CSV format provided should already be in the correct format, so long as you rename the area paths, and delete the custom columns or add the custom fields into your Azure DevOps. You need to keep three columns for the Titles as Alexandar suggests as well. This is how Azure DevOps creates relationships between the rows you are importing, so you should not attempt to combine the three Title columns. Here is an article that explains why you need the multiple Title columns: https://learn.microsoft.com/en-us/azure/devops/boards/queries/import-work-items-from-csv?view=azure-devops#q-can-i-import-a-csv-file-that-has-parent-child-links

**Reply by ClaraSeijoBlanque** [reply]
Hi Rachel,

This is the error that I received and you can see the file I'm trying to import:

![image](https://github.com/microsoft/dynamics365patternspractices/assets/131162800/714cae4a-004b-4545-9bff-56992f82b063)

![image](https://github.com/microsoft/dynamics365patternspractices/assets/131162800/5d285687-6698-4da3-be02-f98862470485)

Thanks for your support,
Clara

**Comment by edupont04** [comment]
We have published an article that talks about how to use the business process catalog to generate work items: https://learn.microsoft.com/en-us/dynamics365/guidance/business-processes/about-import-catalog-devops

You can provide feedback on the article, or submit corrections.

**Comment by ClaraSeijoBlanque** [comment]
Hi all, thanks for your new article, it helps a lot. I found the reason why the import was failing even if I was following the step by step and the reason was the DevOps project setup. Now with the project correct I have not issues at all with the import, this is fantastic, thanks again!!!

Just a small comment for the article, you mentioned to download this file:

![image](https://github.com/microsoft/dynamics365patternspractices/assets/131162800/d851d719-0b88-4e31-83dc-059a66b5844f)


But the correct one to import is this one:

![image](https://github.com/microsoft/dynamics365patternspractices/assets/131162800/d0699a98-4234-457f-86c8-497c94bbac3a)

Thanks again for your help and great/super useful tool.
Best regards

Clara

**Reply by edupont04** [reply]
Hi Clara,
I'm delighted to hear that you find our new article helpful. We'll get the shortlink (aka.ms/BusinessProcessCatalog) updated to point at the October 2023 version later today.

**Reply by ClaraSeijoBlanque** [reply]
Thanks Eva!!

**Reply by ClaraSeijoBlanque** [reply]
Hi Eva,

A few small details about the import that I found during the process:

- I need to open the file in excel several times to do the changes you mentioned and that are required and also to split my process in 1000 lines to be able to import. All ok, the only small issue is the use of comma in the description. This comma are understood as delimiter and  the few lines that include it I need to fix it manually, so, my solution was to replace all the commas before I started to work with the file.

- There is a line that includes quotation mark in the title (_Implement "secret shopper" program_) and this generates an error in DevOps. Easy solution is to remove the " before the import.

Hope this helps. Thanks 
Clara

**Reply by edupont04** [reply]
Thanks for the feedback!

**Reply by stinemarist** [reply]
Hello Clara
I have the same issue as you.. do you remember the reason why the import was failing even if you followed the step by step guide?

Thank you in advance 
Stine

**Reply by ClaraSeijoBlanque** [reply]
Hi @stinemarist 

Yes, I remember. In my case the error was the DevOps Project, it was created wrong at the beginning, this parameter was not selected correctly:

![image](https://github.com/microsoft/dynamics365patternspractices/assets/131162800/3ee389e4-6341-46c5-aee0-de932340693d)

So, we deleted the project and created it again and then all works.

Hope this works in your case.

Regards,
Clara
